packages/client/src/clients/guide/client.ts

```diff
@@ -99,9 +106,17 @@ const select = (state: StoreState, filters: SelectFilterParams = {}) => {
   }
 
   for (const [index, guideKey] of displaySequence.entries()) {
-    const guide = state.guides[guideKey];
+    let guide = state.guides[guideKey];
     if (!guide) continue;
 
+    // Use preview guide if it exists and matches the forced guide key
+    if (
+      state.debug.forcedGuideKey === guideKey &&
+      state.previewGuides[guideKey]
+    ) {
+      guide = state.previewGuides[guideKey];
```

Comment:
It probably doesn't make much difference in reality, but we are swapping out the guide ahead of `predicate()` which means we are filtering in predicate() with the preview guide vs the regular one. Could we move this down below predicate() but above result.set(), so we continue to use the real guide and only swap it out at the end?